### PR TITLE
Python TemplateAccessor and CmdResponseProtocol

### PR DESCRIPTION
**Pull request by ryanmelt**

### CODE REVIEW

File: openc3/python/openc3/accessors/template_accessor.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        self.item_keys = []
+        for item in template_items:
+            self.item_keys.append(item[1:-1])
+            # If they're using parens we have to escape them
+            # since we're working with the already escaped template
+            if self.left_char == "(":
+                item = f"\({item[1:]}"
+            if self.right_char == ")":
+                item = f"{item[0:-1]}\)"
```
Comment (jmthomas) [inline openc3/python/openc3/accessors/template_accessor.py:52]: At line 34 and 37 it is a double `\\` but here a single `\(`. Is that correct? Ruby has double `\\`.

File: openc3/python/openc3/accessors/template_accessor.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        # Scan the response for all the variables in brackets <VARIABLE>
+        values = self.read_regexp.match(buffer.decode())
+        if values is not None:
+            values = values.groups()
+        if values is None or (len(values) != len(self.item_keys)):
+            num_items = 0
+            if values is not None:
+                num_items = len(values)
+            raise RuntimeError(
+                f"Unexpected number of items found in buffer= {num_items}, Expected= {len(self.item_keys)}"
```
Comment (jmthomas) [inline openc3/python/openc3/accessors/template_accessor.py:71]: Ruby code is `buffer:` and `Expected:`

File: openc3/python/openc3/accessors/template_accessor.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
```
Comment (jmthomas) [inline openc3/python/openc3/accessors/template_accessor.py:94]: Change `=` to `:`

**Review by jmthomas — CHANGES_REQUESTED** [review]

File: openc3/python/openc3/accessors/template_accessor.py (hunk 1 above)
Comment (ryanmelt) [reply openc3/python/openc3/accessors/template_accessor.py:52]: Yes.  One is for a regex and the other is for a string replace

File: openc3/python/openc3/accessors/template_accessor.py (hunk 2 above)
Comment (ryanmelt) [reply openc3/python/openc3/accessors/template_accessor.py]: Done

File: openc3/python/openc3/accessors/template_accessor.py (hunk 3 above)
Comment (ryanmelt) [reply openc3/python/openc3/accessors/template_accessor.py]: Done

File: openc3/python/openc3/accessors/template_accessor.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+        escaped_left_char = self.left_char
+        if self.left_char == "(":
+            escaped_left_char = f"\\{self.left_char}"
+        escaped_right_char = self.right_char
+        if self.right_char == ")":
+            escaped_right_char = f"\\{self.right_char}"
+
+        # Convert the template into a Regexp for reading each item
+        template = self.packet.template[:]
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3/python/openc3/accessors/template_accessor.py]: Updated

**Review by jmthomas — APPROVED** [review]

**Pull request merged**